packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -403,7 +453,6 @@ class IOSDeployDebugger {
     try {
       // Detach lldb from the app process.
       _iosDeployProcess?.stdin.writeln('process detach');
-      _debuggerState = _IOSDeployDebuggerState.detached;
```

Comment:
Handle this state only in the log parser so logs can be printed until the app actually detaches.